kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/ResourceMojo.java

```diff
@@ -252,6 +255,10 @@ protected RuntimeMode getRuntimeMode() {
         if (namespace != null && !namespace.isEmpty()) {
             resources = ResourceConfig.toBuilder(resources).namespace(namespace).build();
         }
+        // only overwrite replica count if it hasn't been provided in pom.xml
+        if (replicas != null && ResourceConfig.toBuilder(resources).build().getReplicas() == null) {
```

Comment:
I think one of my [comments](https://github.com/eclipse/jkube/pull/640#issuecomment-810753726) might be misleading and lead you to add the &&... check.

That **shouldn't** be there, because if that's users wouldn't be able to override the value via command line.

```suggestion
        if (replicas != null) {
```

_My comment was more related to the fact of declaring the `replicas` field as an Integer (as opposed to an int) and nullchecking the value (replicas !=null). If the field was to be declared as a primitive type, it would be confusion and difficult to know when the parameter should override the value or not._

Sorry :pray: